static/app/components/commitRow.tsx

```diff
@@ -12,93 +13,85 @@ import TimeSince from 'sentry/components/timeSince';
 import {IconWarning} from 'sentry/icons';
 import {t, tct} from 'sentry/locale';
 import space from 'sentry/styles/space';
-import {Commit, User} from 'sentry/types';
+import {Commit} from 'sentry/types';
 
-type Props = {
+function formatCommitMessage(message: string | null) {
+  if (!message) {
+    return t('No message provided');
+  }
+
+  return message.split(/\n/)[0];
+}
+
+interface CommitRowProps {
   commit: Commit;
   customAvatar?: React.ReactNode;
-};
+}
 
-class CommitRow extends React.Component<Props> {
-  renderMessage(message: Commit['message']): string {
-    if (!message) {
-      return t('No message provided');
+function CommitRow({commit, customAvatar}: CommitRowProps) {
+  const handleInviteClick = React.useCallback(() => {
+    if (!commit.author?.email) {
```

Comment:
This must have somehow escaped the check before, but it seems that because we intersect 
`export type User = Omit<AvatarUser, 'options'> & { type without an email }`, we end up with User type that can have email as undefined...

Reply:
oh good catch 👏👏